pkg/model/node_component_energy.go

```diff
@@ -27,37 +27,46 @@ import (
 	"k8s.io/klog/v2"
 )
 
-var (
-	// the absolute power model includes both the absolute and idle power consumption
-	nodeComponentPowerModel PowerModelInterface
-)
+// the absolute power model includes both the absolute and idle power consumption
+var nodeComponentPowerModel PowerModelInterface
 
 // createNodeComponentPowerModelConfig: the node component power model url must be set by default.
 func createNodeComponentPowerModelConfig(nodeFeatureNames, systemMetaDataFeatureNames, systemMetaDataFeatureValues []string) *types.ModelConfig {
 	modelConfig := CreatePowerModelConfig(config.NodeComponentsPowerKey)
 	if modelConfig.InitModelURL == "" {
 		modelConfig.InitModelFilepath = config.GetDefaultPowerModelURL(modelConfig.ModelOutputType.String(), types.ComponentEnergySource)
+		// HACK: this is hardcoded until model metadata has the trainer information
+		modelConfig.TrainerName = "LogarithmicRegressionTrainer"
```

Comment:
I don't think we should have this hardcode. Prefer to extract model from the file path by having an error check when the file path doesn't contain the trainer name in expected list.

Reply:
The name of the file in question is `intel_rapl_AbsPower.json` and `acpi_AbsPower.json` . is the suggestion to modify the filename to something like `intel_rapl_AbsPower_LogisticRegressionTrainer.json` ?